docs/UpgradeGuide2.md

```diff
@@ -0,0 +1,529 @@
+end
+```
+
+<h3 id="2.0-tracing-api">Tracing API</h3>
+
+Remove option `span_type` from `Datadog::Tracing.trace` method and the following alias methods
+
+| 1.x                                            | Replacement in 2.0                        |
+| ---------------------------------------------- | ----------------------------------------- |
+| `Datadog::Tracing.trace(name, span_type: ...)` | `Datadog::Tracing.trace(name, type: ...)` |
+| `Datadog::Tracing::SpanOperation#span_id`      | `Datadog::Tracing::SpanOperation#id`      |
+| `Datadog::Tracing::SpanOperation#span_type`    | `Datadog::Tracing::SpanOperation#type`    |
+| `Datadog::Tracing::SpanOperation#span_type=`   | `Datadog::Tracing::SpanOperation#type=`   |
+| `Datadog::Tracing::Span#span_id`               | `Datadog::Tracing::Span#id`               |
+| `Datadog::Tracing::Span#span_type`             | `Datadog::Tracing::Span#type`             |
+| `Datadog::Tracing::Span#span_type=`            | `Datadog::Tracing::Span#type=`            |
+
+If you are using [manual instrumentation](GettingStarted.md#manual-instrumentation) or [processing pipeline](GettingStarted.md#processing-pipeline)
+
+```ruby
+# === with 1.x ===
+Datadog::Tracing.trace('my_span', span_type: 'custom') do |span|
+  puts span.span_id
+
+  span.span_type = "...."
+end
+
+# === with 2.0 ===
+Datadog::Tracing.trace('my_span', type: 'custom') do |span|
+  puts span.id
+
+  span.type = "...."
+end
+```
+
+<h3 id="2.0-log-correlation">Log Correlation</h3>
+
+Remove obsolete fields from `Datadog::Tracing::Correlation::Identifier`, and responds to `#env`, `#service`, `#version`, `#trace_id`, `#span_id` and `#to_log_format`. The values returned from `#trace_id` and `#span_id` change from `Integer` to `String`.
+
+If you are [manually correlating logs](GettingStarted.md#trace-correlation), check if it is still compatible.
+
+```ruby
+# === with 1.x ===
+Datadog::Tracing.correlation.span_id
+# => 50288418819650436
+
+# === with 2.0 ===
+Datadog::Tracing.correlation.span_id
+# => '50288418819650436'
+```
+
+<h3 id="2.0-distributed-tracing">Distributed Tracing</h3>
+
+<h4 id="2.0-distributed-tracing-api">Propagation API changes</h4>
+
+If you are [manually propagating distributed tracing metadata](GettingStarted.md#using-the-http-propagator) `Datadog::Tracing::Propagation::HTTP` has moved to `Datadog::Tracing::Contrib::HTTP`.
+
+```ruby
+# === with 1.x ===
+Datadog::Tracing::Propagation::HTTP.inject!
+Datadog::Tracing::Propagation::HTTP.extract
+
+# === with 2.0 ===
+Datadog::Tracing::Contrib::HTTP.inject
+Datadog::Tracing::Contrib::HTTP.extract
+```
+
+<h4 id="2.0-distributed-tracing-env">Environment variable changes</h4>
+
+| 1.x                            | 2.0                                  |
+| ------------------------------ | ------------------------------------ |
+| `DD_PROPAGATION_STYLE_INJECT`  | `DD_TRACE_PROPAGATION_STYLE_INJECT`  |
+| `DD_PROPAGATION_STYLE_EXTRACT` | `DD_TRACE_PROPAGATION_STYLE_EXTRACT` |
+
```

Comment:
Perhaps it is my own ignorance on the topic, so feel free to ignore but it isn't clear to me why values becoming case-insensitive means `the values mapped to different b3 strategies also changed`? Since the values are case-insensitive, it isn't necessary to change existing ones?